logstash-core/lib/logstash/pipeline.rb

```diff
@@ -657,18 +610,25 @@ def shutdown_workers
   def filter(event, &block)
     maybe_setup_out_plugins
     # filter_func returns all filtered events, including cancelled ones
-    filter_func(event).each {|e| block.call(e)}
+    wait_until_started
```

Comment:
@andrewvc I would just add a general issue asking to remove this method and rewrite the tests in `conditionals_spec.rb` accordingly. I think a todo of this "magnitude" will just go ignored forever :D?